Bottom: e0e829af3daeba6119c1f7801e924a5fc314f7dc
Top:    398068a27cc9f013e6106f95f43243416468bd4d
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-07 01:38:04 +0000

Refresh of remove-setdefaults-parameter

---

diff --git a/src/action.c b/src/action.c
index c886d79..88ac6cc 100644
--- a/src/action.c
+++ b/src/action.c
@@ -5939,7 +5939,7 @@ ActionNew (int argc, char **argv, Coord x, Coord y)
       RemovePCB (PCB);
       PCB = NULL;
       PCB = CreateNewPCB ();
-      CreateNewPCBPost (PCB);
+      CreateNewPCBPost (PCB, 1);
 
       /* setup the new name and reset some values to default */
       free (PCB->Name);
diff --git a/src/create.c b/src/create.c
index 1f85ca6..fe9619a 100644
--- a/src/create.c
+++ b/src/create.c
@@ -220,7 +220,7 @@ CreateNewPCB (void)
  * pre-existing PCB.
  */
 int
-CreateNewPCBPost (PCBType *pcb)
+CreateNewPCBPost (PCBType *pcb, int use_defaults)
 {
   /* copy default settings */
   pcb_colors_from_settings (pcb);
diff --git a/src/create.h b/src/create.h
index f352e06..f49f6d0 100644
--- a/src/create.h
+++ b/src/create.h
@@ -42,7 +42,7 @@ void pcb_colors_from_settings (PCBType *);
 PCBType * CreateNewPCB ();
 /* Called after PCB->Data->LayerN is set.  Returns zero if no errors,
    else nonzero.  */
-int CreateNewPCBPost (PCBType *);
+int CreateNewPCBPost (PCBType *, int /* set defaults */);
 PinType * CreateNewVia (DataType *, Coord, Coord, Coord, Coord, Coord, Coord, char *, FlagType);
 LineType * CreateDrawnLineOnLayer (LayerType *, Coord, Coord, Coord, Coord, Coord, Coord, FlagType);
 LineType * CreateNewLineOnLayer (LayerType *, Coord, Coord, Coord, Coord, Coord, Coord, FlagType);
diff --git a/src/main.c b/src/main.c
index 673a7b9..3c79bac 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1993,7 +1993,7 @@ main (int argc, char *argv[])
   PCB = CreateNewPCB ();
   ParseGroupString (Settings.Groups, &PCB->LayerGroups, &PCB->Data->LayerN);
   /* Add silk layers to newly created PCB */
-  CreateNewPCBPost (PCB);
+  CreateNewPCBPost (PCB, 1);
   if (argc > 1)
     command_line_pcb = argv[1];
